pkg/index/validation/validate.go

```diff
@@ -124,6 +124,9 @@ func validateSelector(sel *metav1.LabelSelector) error {
 	if sel == nil {
 		return errors.New("nil selector is not supported")
 	}
+	if sel.MatchLabels == nil && len(sel.MatchExpressions) == 0 {
```

Comment:
To be on the safe side, this should be
```go
if len(sel.MatchLabels) == 0 && len(sel.MatchExpressions) == 0 {
```

Otherwise, `sel.MatchLabels` could be non-nil, but empty. And that would match everything, right?

Reply:
we have a check for that case below?

Reply:
Oh, I see.